include/eosio/bitset.hpp

```diff
@@ -0,0 +1,194 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE
+ */
+#pragma once
+
+#include "from_bin.hpp"
+#include "from_json.hpp"
+#include "to_bin.hpp"
+#include "to_json.hpp"
+
+namespace eosio {
+
+// -------------------------------------------------------------------------------
+//      see https://github.com/AntelopeIO/spring/wiki/ABI-1.3:-bitset-type
+// -------------------------------------------------------------------------------
+
+
+// stores a bitset in a std::vector<uint8_t>
+//
+// - bits 0-7 in first byte, 8-15 in second, ...
+// - least significant bit of byte 0 is bit 0 of bitset.
+// - unused bits must be zero.
+// ---------------------------------------------------------------------------------
+struct bitset {
+   using buffer_type                         = std::vector<uint8_t>;
+   using size_type                           = uint32_t;
+   static constexpr size_type bits_per_block = 8;
+   static constexpr size_type npos           = static_cast<size_type>(-1);
+
+   static constexpr size_type calc_num_blocks(size_type num_bits) {
+      return (num_bits + bits_per_block - 1) / bits_per_block;
+   }
+
+   static size_type block_index(size_type pos) noexcept { return pos / bits_per_block; }
+   static uint8_t   bit_index(size_type pos)   noexcept { return static_cast<uint8_t>(pos % bits_per_block); }
+   static uint8_t   bit_mask(size_type pos)    noexcept { return uint8_t(1) << bit_index(pos); }
+
+   size_type size() const { return m_num_bits; }
+
+   void resize(size_type num_bits) {
+      m_bits.resize(calc_num_blocks(num_bits), 0);
+      m_num_bits = num_bits;
+      zero_unused_bits();
+   }
+
+   void clear() {
+      for (auto& byte : m_bits)
+         byte = 0;
+   }
```

Comment:
This isn't the same behavior of boost's dynamic_bitset -- `clear()` there changes the size to 0. Not saying we need to match but personally I would have guessed boost's behavior over this.

Reply:
Agreed, somewhat misleading. I'll change the name.

Reply:
https://github.com/AntelopeIO/abieos/pull/47/commits/34779c455e6114107b190bda6bb197bc0bc39531